monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppHelper.java

```
@@ -32,9 +39,15 @@ public IApp copyAppInfo(IApp app, AppForm appForm) {
         if (appForm.getRecipientIds() != null) {
             app.setRecipients(getRecipientsById(appForm.getRecipientIds()));
         }
+        if (appForm.getGroupType()==GroupType.NEW) {
+            app.setGroup(appManager.createGroup(appForm.getGroupId()));
+        }
+        else {
+            app.setGroup(appManager.getGroup(appForm.getExistingGroupId()));
```

what if the group doesn't exist? you might want to throw an exception.